cmd/api/src/api/v2/cypherquery.go

```diff
@@ -60,69 +59,69 @@ func (s Resources) CypherSearch(response http.ResponseWriter, request *http.Requ
-			)
-			return
-		}
-
-		// All mutation attempts must be audit logged even when failed
-		if auditLogEntry, err = model.NewAuditEntry(
-			model.AuditLogActionMutateGraph,
-			model.AuditLogStatusIntent,
-			model.AuditData{"query": preparedQuery.StrippedQuery}); err != nil {
-			api.WriteErrorResponse(
-				request.Context(),
-				api.BuildErrorResponse(http.StatusInternalServerError, api.ErrorResponseDetailsInternalServerError, request),
-				response,
-			)
-			return
-		}
+		graphResponse, err = s.cypherMutation(request, preparedQuery, payload.IncludeProperties)
+	} else {
+		graphResponse, err = s.GraphQuery.RawCypherQuery(request.Context(), preparedQuery, payload.IncludeProperties)
+	}
 
-		// create an intent audit log
-		if err := s.DB.AppendAuditLog(request.Context(), auditLogEntry); err != nil {
+	if err != nil {
+		switch true {
+		case strings.Contains(err.Error(), "Permission denied: User may not modify the graph."):
 			api.WriteErrorResponse(
 				request.Context(),
-				api.BuildErrorResponse(http.StatusInternalServerError, api.ErrorResponseDetailsInternalServerError, request),
-				response,
+				api.BuildErrorResponse(http.StatusForbidden, err.Error(), request), response,
 			)
-			return
-		}
-	}
-
-	if graphResponse, err := s.GraphQuery.RawCypherSearch(request.Context(), preparedQuery, payload.IncludeProperties); err != nil {
-		// audit log needs to be marked as failed if a mutation is present
-		if preparedQuery.HasMutation {
-			auditLogEntry.Status = model.AuditLogStatusFailure
-		}
-		if queries.IsQueryError(err) {
+		case queries.IsQueryError(err):
 			api.WriteErrorResponse(
 				request.Context(),
 				api.BuildErrorResponse(http.StatusBadRequest, err.Error(), request), response,
 			)
-		} else if util.IsNeoTimeoutError(err) {
+		case util.IsNeoTimeoutError(err):
 			api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusInternalServerError, "transaction timed out, reduce query complexity or try again later", request), response)
-		} else {
+		default:
 			api.WriteErrorResponse(
 				request.Context(),
 				api.BuildErrorResponse(http.StatusInternalServerError, err.Error(), request), response,
 			)
 		}
 	} else {
-		// audit log needs to be marked as success if a mutation is present
-		if preparedQuery.HasMutation {
-			auditLogEntry.Status = model.AuditLogStatusSuccess
-		}
 		api.WriteBasicResponse(request.Context(), graphResponse, http.StatusOK, response)
 	}
+}
 
-	if preparedQuery.HasMutation {
-		if err := s.DB.AppendAuditLog(request.Context(), auditLogEntry); err != nil {
-			// We don't want to update response as failed because having info on the mutation response trumps this error
-			log.Errorf("failure to create mutation audit log %s", err.Error())
-		}
+func (s Resources) cypherMutation(request *http.Request, preparedQuery queries.PreparedQuery, includeProperties bool) (model.UnifiedGraph, error) {
```

Comment:
I think it might make sense to pull the authCtx off of request before sending it into this function. That way the function no longer needs to worry about request semantics, but just the AuthCtx for checking permissions.

Reply:
I guess we'd also want to grab a context off of request and pass it down here too, but that's not a bad idea as it helps ensure the context requirement is obvious

Reply:
I gave this a try actually but reverted because I then have to update the `AuditLogUnauthorizedAccess` to take in the 3 request parts it is concerned about and that ballooned out the arguments so I eventually embraced the request argument and nixed the authCtx entirely. I can add it back if it makes it more readable?

Reply:
Yeah, that's fine then. Was just maybe a way for us to clean up dependencies for later refactors, but not necessary